tests/anoph/conftest.py

```diff
@@ -584,6 +584,318 @@ def simulate_aim_variants(path, contigs, snp_sites, n_sites_low, n_sites_high):
+    root = zarr.open(zarr_path, mode="w")
+
+    # Create samples array.
+    df_samples = pd.read_csv(metadata_path)
+    n_samples = len(df_samples)
+    samples = df_samples["sample_id"].values
+    root.create_dataset(name="samples", data=samples, dtype=str)
+
+    for contig in contigs:
+        # Create the contig group.
+        contig_grp = root.require_group(contig)
+
+        # Create the calldata group for this contig.
+        calldata = contig_grp.require_group("calldata")
+
+        # Get the SNP positions for this contig.
+        snp_pos = snp_sites[f"{contig}/variants/POS"][:]
+
+        # Get the number of sites (SNP positions) for this contig.
+        n_sites = snp_pos.shape[0]
+
+        # Simulate the genotype calls.
+        # Note: this is only 2D, unlike SNP, HAP, AIM GT which are 3D
+        gt = np.random.choice(
+            np.array([0, 1], dtype="i1"),
+            size=(n_sites, n_samples),
+            replace=True,
+            p=[1 - p_allele, p_allele],
+        )
+
+        # Create the GT dataset under calldata.
+        calldata.create_dataset(name="GT", data=gt)
+
+        # Create the samples dataset (again) for this contig.
+        contig_grp.create_dataset(name="samples", data=samples, dtype=str)
+
+        # Create the variants group for this contig.
+        variants_grp = contig_grp.require_group("variants")
+
+        # Simulate the CIEND and CIPOS arrays under variants.
+        ciend = np.random.randint(low=0, high=13200, size=n_sites)
+        cipos = np.random.randint(low=0, high=37200, size=n_sites)
+        variants_grp.create_dataset(name="CIEND", data=ciend)
+        variants_grp.create_dataset(name="CIPOS", data=cipos)
+
+        # Simulate the unique ID strings under variants.
+        # Note: this is quicker than generating unique random strings.
+        len_str_n_sites = len(str(n_sites))
+        variant_IDs = [
+            f"CNV_{contig}{str(i).zfill(len_str_n_sites)}"
+            for i in range(1, n_sites + 1)
+        ]
+        variants_grp.create_dataset(name="ID", data=variant_IDs)
+
+        # Simulate the filters under variants.
+        filter_pass = np.random.choice(
+            [False, True], size=n_sites, p=[1 - p_filter_pass, p_filter_pass]
+        )
+        filter_qMerge = np.random.choice(
+            [False, True], size=n_sites, p=[1 - p_filter_qMerge, p_filter_qMerge]
+        )
+        variants_grp.create_dataset(name="FILTER_PASS", data=filter_pass)
+        variants_grp.create_dataset(name="FILTER_qMerge", data=filter_qMerge)
+
+        # Simulate POS under variants.
+        # Note: choosing a random subset would cause dimension size conflicts.
+        variants_grp.create_dataset(name="POS", data=snp_pos)
+
+        # Simulate END under variants.
+        random_multipliers = np.random.randint(2, 12, size=len(snp_pos))
+        end = random_multipliers * snp_pos
+        variants_grp.create_dataset(name="END", data=end)
```

Comment:
See comment above regarding POS and END for coverage calls.

Reply:
Thanks. Have done.